src/asm_files.cpp

```diff
@@ -26,9 +26,15 @@ static vector<T> vector_of(ELFIO::section* sec) {
 }
 
 int create_map_crab(uint32_t map_type, uint32_t key_size, uint32_t value_size, uint32_t max_entries, ebpf_verifier_options_t options) {
-    // For now we just make up a number as if the map were created,
-    // without actually creating anything.
-    return (value_size << 14) + (key_size << 6); // + i;
+    if (map_type == 12 || map_type == 13) {
+        return -1;
+    }
+    if (key_size > (1 << 8))
+        throw std::runtime_error("bad map key size " + std::to_string(key_size));
+    if (value_size > (1 << (31 - 8)))
+        throw std::runtime_error("bad map value size " + std::to_string(value_size));
+    int res = (value_size << 8) + key_size;
```

Comment:
I'm not sure there's any reason left to embed these numbers in the mock fd.  Maybe we should just use a sequence number.

Reply:
Using static state failed tests. I'll use something more appropriate.

Reply:
The code is probably ok as is for the moment.

Reply:
I'll merge for now (after signing-off).